### PR TITLE
Feature/search accommodations

### PR DESCRIPTION
**Pull request by zilola08**
- added search functionality;

### CODE REVIEW

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx [hunk 1]
```diff
@@ -79,56 +177,113 @@ const SearchModal = ({ modalOpen, changeOpen }: ExploreModalProps) => {
...
-      </SafeAreaView>
+        <FlatList
+          data={sections}
+          keyExtractor={(item) => item.type}
+          renderItem={({ item }) => {
+            if (item.type === 'location') {
+              return (
+                <Collapsable
+                  title={item.title}
+                  subtitle={location ? location : "Anywhere"}
```
Comment (gitkoshelew) [inline src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: you already have constant for 'Anywhere', so use it, just transver to constants file

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx [hunk 2]
```diff
@@ -79,56 +177,113 @@ const SearchModal = ({ modalOpen, changeOpen }: ExploreModalProps) => {
...
+                <Collapsable
+                  title={item.title}
+                  subtitle={checkOutDate ? checkOutDate : "Anyday"}
+                  contentTitle={item.title}
+                  collapsed={isCollapsed.checkout}
+                  onTouch={toggleCollapseCheckOut}
+                >
+                  <DatePicker
+                    options={{
+                      backgroundColor: colors ? BLACK : WHITE,
```
Comment (gitkoshelew) [inline src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: what do you think to create datepickerOptions memoized value like

const DatePickerStyles = usememo(() => {
 const options = {
                      selectedTextColor: WHITE,
                      mainColor: TOMATO,
                      textSecondaryColor: GREY_200,
}
const colorOption = color ? {backgroundColor: BLACK, textHeaderColor: WHITE, ...} : 
return {...option, ...colorOptions}
), [color]

moreOver, you use it twice in one file

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.utils.ts [hunk 3]
```diff
@@ -1,19 +1,95 @@
...
+
+  const formattedLocationString = `${city}, ${country}`;
+  return formattedLocationString;
+};
+
+export const isInvalidDateRange = (checkin: string, checkout: string) => {
+  if (checkin === '' && checkout === '') return false;
+  const today = new Date().setHours(0, 0, 0, 0);;
+  const checkInTime = new Date(checkin.replace(/\//g, '-')).setHours(0, 0, 0, 0);
+  const checkOutTime = new Date(checkout.replace(/\//g, '-')).setHours(0, 0, 0, 0);
```
Comment (gitkoshelew) [inline src/components/Modals/ExploreModals/SearchModal/SearchModal.utils.ts]: create file for regular expressions and save it there, and use here to not repeat yourself

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx [hunk 4]
```diff
@@ -1,63 +1,161 @@
...
-    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.wherePressed);
+  const toggleCollapseCheckIn = () => {
+    isCollapsed.checkin ? 
+    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.checkInPressed) :
+    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.allClosed);
   };
+
+  const toggleCollapseCheckOut = () => {
+    isCollapsed.checkout ? 
+    setIsCollapsed(COLLAPSABLE_CARDS_POSITIONS.checkOutPressed) :
```
Comment (gitkoshelew) [inline src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: if you need to change related values several times inside a component, better to use useReducer, not mandatory, just keep in mind

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx (hunk 1 above)
Comment (zilola08) [reply src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: done

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx (hunk 2 above)
Comment (zilola08) [reply src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: thanks, done

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.utils.ts (hunk 3 above)
Comment (zilola08) [reply src/components/Modals/ExploreModals/SearchModal/SearchModal.utils.ts]: sure, done

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx (hunk 4 above)
Comment (zilola08) [reply src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: sure, moved toggling all collapsable sections from useState to useReducer

**Review by gitkoshelew — APPROVED** [review]

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx [hunk 5]
```diff
@@ -1,63 +1,192 @@
...
 };
 
+const ActionTypes = {
+  TOGGLE_LOCATION_SECTION: 'TOGGLE_LOCATION_SECTION',
+  TOGGLE_CHECKIN_SECTION: 'TOGGLE_CHECKIN_SECTION',
+  TOGGLE_CHECKOUT_SECTION: 'TOGGLE_CHECKOUT_SECTION',
+}
+
+const reducer = (state: any, action: any) => {
+  switch (action.type) {
```
Comment (javokhirm) [inline src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: don't write state logic here inside the screen, instead, you can create an additional `modalSlice`

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx [hunk 6]
```diff
@@ -79,56 +208,97 @@ const SearchModal = ({ modalOpen, changeOpen }: ExploreModalProps) => {
...
-          </Collapsable>
-        </View>
-      </SafeAreaView>
+        <FlatList
+          data={sections}
+          keyExtractor={(item) => item.type}
+          renderItem={({ item }) => {
+            if (item.type === 'location') {
+              return (
+                <Collapsable
```
Comment (javokhirm) [inline src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: let's make these lines more readable, you can make separate collapsible components like `CollapsibleLocationItem`, `CollapsibleCheckInItem`...

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx [hunk 7]
```diff
@@ -1,63 +1,192 @@
...
     <Modal
       animationType="fade"
       visible={modalOpen}
       onRequestClose={() => {
         changeOpen();
-      }}
+      } }
     >
       <StatusBar backgroundColor={colors ? BLACK : WHITE_100} />
       <SafeAreaView
```
Comment (javokhirm) [inline src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx:137]: we already have `ScreenTemplate` component to wrap screens with safe area view, can we use it here?

File: src/utils/index.ts [hunk 8]
```diff
@@ -2,3 +2,4 @@ export * from './ui';
 export * from './validation';
 export * from './languages';
 export * from './getQuery';
+export * from './stringToDateFormat';
```
Comment (javokhirm) [inline src/utils/index.ts]: let's rename this file to `date.ts`, we may have another date utils as well

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx (hunk 5 above)
Comment (zilola08) [reply src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: thanks, moved reducer logic to slice

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx (hunk 6 above)
Comment (zilola08) [reply src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx]: sure, done

File: src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx (hunk 7 above)
Comment (zilola08) [reply src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx:137]: yes, done

File: src/utils/index.ts (hunk 8 above)
Comment (zilola08) [reply src/utils/index.ts]: done

**Pull request merged**